File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.java
class Solution {
    public int solution(int n, int a, int b) {
        for(int round = 1; round <= Math.log(n) / Math.log(2); round++) {
            if (Math.abs(a - b) == 1 && Math.min(a, b) % 2 == 1) return round;
            a = (a + 1) / 2;
            b = (b + 1) / 2;
        }
        throw new RuntimeException("The game is over");
    }
}

File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.kt
import java.lang.RuntimeException
import kotlin.math.abs
import kotlin.math.log2
import kotlin.math.min

class Solution {
    fun solution(n: Int, a: Int, b: Int): Int = (1..log2(n.toDouble()).toInt()).fold(Pair(a, b)) { nums, round ->
        if (abs(nums.first - nums.second) == 1 && min(nums.first, nums.second) % 2 == 1) return round
        Pair((nums.first + 1) / 2, (nums.second + 1) / 2)
    }.let { throw RuntimeException("The game is over") }
}